### PR TITLE
feat(parachain/collator): handle backed overseer message

### LINKED ISSUE #3517: feat(parchain/collator): handle Backed message
## Issue summary

- This is part of collator protocol. 
- Collator Protocol Subsystem receives a Backed message from overseer (https://github.com/paritytech/polkadot-sdk/blob/f4827dcd16fc5585a8d15c6a32bc27ecde00e7f9/polkadot/node/network/collator-protocol/src/validator_side/mod.rs#L1525)
- Check if we need to process this message, if so do it.

- This message transfer is between two subsystems on a single node, so we have freedom on its structure.
- Should be done after https://github.com/ChainSafe/gossamer/pull/3512 is merged.

## Other information and links
Code reference in polkadot sdk

https://github.com/paritytech/polkadot-sdk/blob/f4827dcd16fc5585a8d15c6a32bc27ecde00e7f9/polkadot/node/network/collator-protocol/src/validator_side/mod.rs#L1525

Make sure you follow the polkadot-sdk repo. The code has changed enough from the archived polkadot repo.

### PR DESCRIPTION
**Pull request by kishansagathiya**
## Changes

Collator Protocol Subsystem receives a Backed message from overseer. This message tells collator protocol when a candidate has been backed.
This PR processes that backed message. Unblocks blocked advertisements as per the situation.

## Tests

```sh
go test -tags integration github.com/ChainSafe/gossamer
```

## Issues

Fixes #3517 

## Primary Reviewer

@axaysagathiya

### CODE REVIEW

File: dot/parachain/collator-protocol/validator_side.go [hunk 1]
```diff
@@ -330,11 +330,15 @@ type CollatorProtocolValidatorSide struct {
...
-	// TODO: In rust this is a map, let's see if we can get away with a map
-	// blocked_advertisements: HashMap<(ParaId, Hash), Vec<BlockedAdvertisement>>,
-	BlockedAdvertisements []BlockedAdvertisement
+	// Advertisements that were accepted as valid by collator protocol but rejected by backing.
+	//
+	// It's only legal to fetch collations that are either built on top of the root
+	// of some fragment tree or have a parent node which represents backed candidate.
+	// Otherwise, a validator will keep such advertisement in the memory and re-trigger
+	// requests to backing on new backed candidates and activations.
+	BlockedAdvertisements map[string][]BlockedAdvertisement
```
Comment (axaysagathiya) [inline dot/parachain/collator-protocol/validator_side.go:341]: Do you need this field exported?

File: dot/parachain/collator-protocol/validator_side.go [hunk 2]
```diff
@@ -372,6 +376,21 @@ func (f fetchedCollationInfo) String() string {
...
+type ProspectiveParachainsMode struct {
+	// if disabled, there are no prospective parachains. Runtime API does not have support for `async_backing_params`
+	isEnabled bool
+
+	// these values would be present only if `isEnabled` is true
+
+	// The maximum number of para blocks between the para head in a relay parent and a new candidate.
+	// Restricts nodes from building arbitrary long chains and spamming other validators.
+	maxCandidateDepth uint
+}
```
Comment (axaysagathiya) [inline dot/parachain/collator-protocol/validator_side.go:392]: We already have this type in `dot/parachain/types/types.go`

File: dot/parachain/collator-protocol/validator_side.go (hunk 1 above)
Comment (kishansagathiya) [reply dot/parachain/collator-protocol/validator_side.go]: addressed in latest commit

**Review by axaysagathiya — APPROVED** [review]
LGTM

**Review by EclesioMeloJunior — APPROVED** [review]

**Pull request merged**